src/service/viewer-impl.js

```diff
@@ -802,6 +802,25 @@ export class Viewer {
   }
 
   /**
+   * Get the fragment from the url or the viewer.
+   * Strip leading '#' in the fragment
+   * @return {!Promise<string|undefined>}
+   */
+  getFragment() {
+    if (!this.isEmbedded_) {
```

Comment:
Ok. Generally, Viewer has access to it hash before  opening up AMP. But, your solution is better since it's more flexible - it'd actually allow iframe construction on the server side if anyone ever desires that. So, let's keep your way.